src/DotNetWorker.Core/Invocation/FunctionInvocationManager.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Concurrent;
+
+namespace Microsoft.Azure.Functions.Worker.Invocation
+{
+    internal class FunctionInvocationDictionary : IFunctionInvocationDictionary
+    {
+        private ConcurrentDictionary<string, FunctionInvocationDetails> _inflightInvocations;
+
+        public FunctionInvocationDictionary()
+        {
+          _inflightInvocations = new ConcurrentDictionary<string, FunctionInvocationDetails>();
+        }
+
+        public bool TryAddInvocationDetails(string invocationId, FunctionInvocationDetails details)
+        {
+            if (string.IsNullOrEmpty(invocationId) || details is null)
+            {
+              return false;
+            }
+
+            return _inflightInvocations.TryAdd(invocationId, details);
+        }
+
+        public bool TryRemoveInvocationDetails(string invocationId)
+        {
+            if (string.IsNullOrEmpty(invocationId))
+            {
+              return false;
```

Comment:
nit: indentation (have 2 spaces here, should be 4). Also the other if statements in this file.